syft/frameworks/torch/tensors/interpreters/replicative_shared.py

```diff
@@ -0,0 +1,65 @@
+import torch
+from syft.generic.abstract.tensor import AbstractTensor
+import random
+
+
+class ReplicatedSharingTensor(AbstractTensor):
+    def __init__(
+        self, shares=None, owner=None, id=None, tags=None, description=None,
+    ):
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.child = shares
+        self.ring_size = 2 ** 5
+
+    def share_secret(self, secret, workers):
+        number_of_shares = len(workers)
+        shares = self.generate_shares(secret, number_of_shares)
+        shares_locations = self.distribute_shares(workers, shares)
+        self.child = shares_locations
+        return self
+
+    def generate_shares(self, secret, number_of_shares=3):
+        shares = []
+        for _ in range(number_of_shares - 1):
+            shares.append(torch.tensor(random.randrange(self.ring_size)))
+        shares.append(torch.tensor(secret - sum(shares) % self.ring_size))
+        return shares
+
+    @staticmethod
+    def distribute_shares(workers, shares):
+        shares_locations = {}
+        assert len(workers) == len(shares)
+        for i in range(len(shares)):
+            pointer1 = shares[i].send(workers[i])
```

Comment:
Q: Shouldn't here the party[i] hold shares i-1 and i + 1?

Reply:
No, it should know shares i and i+1. if it knows i-1 and i+1 it can reconstruct the secret.
section 3.1 of the paper states: "(x1, x2)
is held by P1, (x2, x3) by P2, and (x3, x1) by P3"